Update changelog for v10.29.0 (#18668)
MetaMask/metamask-extension: CHANGELOG.md

## [10.29.0]
### Uncategorized

> - [FLASK] Update snap delineator ([#18385](https://github.com/MetaMask/metamask-extension/pull/18385))

Reviewer: ```suggestion
- [FLASK] Redesign snap content delineator ([#18385](https://github.com/MetaMask/metamask-extension/pull/18385))
```

> - [FLASK] `snaps-monorepo@0.32.2` ([#18371](https://github.com/MetaMask/metamask-extension/pull/18371))

Reviewer: ```suggestion
- [FLASK] **BREAKING:** Snaps are now required to request permission for at least one handler permission (e.g. `onRpcRequest`) ([#18371](https://github.com/MetaMask/metamask-extension/pull/18371))
- [FLASK] Fix issues with using `atob` and `btoa` in snaps ([#18371](https://github.com/MetaMask/metamask-extension/pull/18371))
```

> - [FLASK] Update iframe-execution-environment ([#18299](https://github.com/MetaMask/metamask-extension/pull/18299))

Reviewer: ```suggestion
```

> - [FLASK] `snaps-monorepo@0.31.0` ([#18142](https://github.com/MetaMask/metamask-extension/pull/18142))

Reviewer: ```suggestion
- [FLASK] Combine the snap installation popups into a single popup ([#18142](https://github.com/MetaMask/metamask-extension/pull/18142))
- [FLASK] **BREAKING:** Disallow snaps requesting `eth_requestAccounts` and `wallet_requestSnaps` RPC methods ([#18142](https://github.com/MetaMask/metamask-extension/pull/18142))
```

> - [FLASK] Add hardcoded derivation paths ([#18176](https://github.com/MetaMask/metamask-extension/pull/18176))

Reviewer: ```suggestion
```

### Added
- Adding a new terms of use, displayed during onboarding and in a popover ([#18221](https://github.com/MetaMask/metamask-extension/pull/18221))

> - [FLASK] Add permission cell component ([#18372](https://github.com/MetaMask/metamask-extension/pull/18372))

Reviewer: ```suggestion
- [FLASK] Redesign snaps permission screens ([#18372](https://github.com/MetaMask/metamask-extension/pull/18372))
```

### Changed
- Allow users to go from the send screen to the confirm screen in cases when they do not have enough funds for gas ([#18554](https://github.com/MetaMask/metamask-extension/pull/18554))
- Add Ledger instructions to the Sign In With Ethereum page ([#18589](https://github.com/MetaMask/metamask-extension/pull/18589))
- Improve security provider warning messaging, to give users more info about transactions security providers flag as potentially suspicious ([#18545](https://github.com/MetaMask/metamask-extension/pull/18545))
- Disable account and network switching on the send screen ([#18086](https://github.com/MetaMask/metamask-extension/pull/18086))
- Update wording on token allowance screen: replace "contract" with "third party" ([#18101](https://github.com/MetaMask/metamask-extension/pull/18101))
- Modify the send icon ([#18411](https://github.com/MetaMask/metamask-extension/pull/18411))
- Adds more info to the Chrome and Firefox extensions list about what your build type is ([#18391](https://github.com/MetaMask/metamask-extension/pull/18391))
- [FLASK] Redesign key management modal ([#18263](https://github.com/MetaMask/metamask-extension/pull/18263))

> - [FLASK] Update snap authorship component ([#18262](https://github.com/MetaMask/metamask-extension/pull/18262))

Reviewer: ```suggestion
- [FLASK] Redesign snap authorship component ([#18262](https://github.com/MetaMask/metamask-extension/pull/18262))
```

> - [FLASK] Update UI related to MetaMask Snaps Platform trademark ([#18172](https://github.com/MetaMask/metamask-extension/pull/18172))

Reviewer: ```suggestion
- [FLASK] Improve design of snaps settings page when no snaps are installed ([#18172](https://github.com/MetaMask/metamask-extension/pull/18172))
```

- [FLASK] Remove permission footer in snap install/update flow ([#18240](https://github.com/MetaMask/metamask-extension/pull/18240))

### Fixed
- Fix NFT added notification close button ([#18504](https://github.com/MetaMask/metamask-extension/pull/18504))
- Add a title to the security provider "What's New" notification ([#18526](https://github.com/MetaMask/metamask-extension/pull/18526))
- Fix layout/styling of the "Hold to reveal" button in the SRP reveal flow([#18496](https://github.com/MetaMask/metamask-extension/pull/18496))
- [FLASK] Fix crash when requesting unknown snap permission ([#18447](https://github.com/MetaMask/metamask-extension/pull/18447))

> - [FLASK] Fix overflow issues with text coming from snap UI ([#18169](https://github.com/MetaMask/metamask-extension/pull/18169))

Reviewer: ```suggestion
- [FLASK] Fix window overflow issues with snap UI text ([#18169](https://github.com/MetaMask/metamask-extension/pull/18169))
```
